org.eclipse.xtext.builder/src/org/eclipse/xtext/builder/clustering/CurrentDescriptions.java

```diff
@@ -26,14 +26,16 @@
  * @author Sebastian Zarnekow - Initial contribution and API
  * @author Thomas Wolf <thomas.wolf@paranor.ch> - Performance optimization and JavaDoc
  */
-public class CurrentDescriptions extends AdapterImpl implements IResourceDescriptions {
+public class CurrentDescriptions extends AdapterImpl implements IResourceDescriptions.IResourceSetAware {
 
 	/**
 	 * New index.
 	 */
 	private final ResourceDescriptionsData newData;
 	
 	private BuildData buildData;
+
+	private ResourceSet resourceSet;
```

Comment:
I'm not terribly sure, but AdapterImpl.getTarget already yields the resource set. So calling that one and casting should be good enough.

Reply:
Ok, I will change to using getTarget()! Should I add null-checks and/or cast-checks while doing that to make sure that getTarget() is not null or another type?

Reply:
I think a simple cast but an explicit Preconditions.checkNotNull() would be good to have, yes. 👍

Reply:
Will do, busy days ahead but will squeeze it in somewhere. :-)